app/models/species.rb

```diff
@@ -13,4 +13,15 @@ class Specie < ApplicationRecord
 
   validates :wound_threshold, :strain_threshold, numericality: { only_integer: true, greater_than_or_equal_to: 0,
                                                                  less_than_or_equal_to: 99 }
-end
\ No newline at end of file
+
+  scope :with_name, ->(name) { where(name: name) }
+
+  def self.upload(json_file)
+    JsonSpecies.to_a(json_file).each do |json_specie|
+      json_specie_attrs = json_specie.attributes
+      Species.with_name(json_specie.name).first_or_create(json_specie_attrs).update(json_specie_attrs)
+    end
+  end
```

Comment:
We created the SpeciesUploader

```ruby
class SpeciesUploader

  def initialize json_file
    @json_file = json_file
  end

  def run
    JsonSpecies.to_a(@json_file).each do |json_specie|
      json_specie_attrs = json_specie.attributes
      Species.with_name(json_specie.name).first_or_create(json_specie_attrs).update(json_specie_attrs)
    end
  end
end
```
But reek got us this warning

```
app/services/species_uploader.rb -- 1 warning:
  [9, 10]:FeatureEnvy: SpeciesUploader#run refers to 'json_specie' more than self (maybe move it to another class?) [https://github.com/troessner/reek/blob/master/docs/Feature-Envy.md]
1 total warning
```

But if we used this it doesn't apply.

```ruby
class SpeciesUploader

  def initialize json_file
    @json_file = json_file
  end

  def run
    JsonSpecies.to_a(@json_file).each do |json_specie|
      json_specie_attrs = json_specie.attributes
      Species.with_name(json_specie_attrs.name).first_or_create(json_specie_attrs).update(json_specie_attrs)
    end
  end
end
```